core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java

```diff
@@ -47,55 +47,48 @@
   private int numRecords;
   private int numRecordsRemaining;
 
-  private byte[] arr = new byte[1024 * 1024];
+  private byte[] arr = new byte[1024];
```

Comment:
If this number is performance-senstive, could we parameterize it?

Reply:
Yes. I am looking into this. Thank you

Reply:
Does this look good? Perhaps you have some suggestion.

private[spark] val UNSAFE_SORTER_SPILL_READER_BUFFER_SIZE_RATIO =
    ConfigBuilder("spark.unsafe.sorter.spill.reader.buffer.size.ratio")
      .doc("The multiplication ratio is the parameter that controls the initial read buffer " +
        "size. The multiplication ratio value range is from 1 to 1024. This parameter increases "  +
        "the initial read buffer size in 1KB increments. It will result in the initial buffer " +
        "size in the range from 1KB to 1MB. The read buffer size is dynamically adjusted " +
        "afterward based on data length read from the spilled file.")
      .intConf
      .checkValue(v => 1 <= v && v <= DEFAULT_BUFFER_SIZE_RATIO,
        s"The value must be in allowed range [1, ${DEFAULT_BUFFER_SIZE_RATIO}].")
      .createWithDefault(DEFAULT_BUFFER_SIZE_RATIO)

Reply:
Could you apply the change into this PR, first? Otherwise, its hard to leave comments line-by-line...

Reply:
NP. I will push update soon. I am testing changes with 10TB benchmarks.